src/app/dao/userDao.ts

```diff
@@ -0,0 +1,338 @@
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  static async updatePassword({
+    email,
+    hash,
+    salt,
+  }: {
+    email: string;
+    hash: string;
+    salt: string;
+  }) {
+    if (!this.userCollection) {
+      return;
+    }
+
+    try {
+      const result = await this.userCollection.updateOne(
+        {
+          email,
+        },
+        {
+          $set: {
+            hash,
+            salt,
+          },
+        }
+      );
+      return result;
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  static async allPlans(email: string) {
+    if (!this.userCollection) {
+      return;
+    }
+    try {
+      const cursor = await this.userCollection.aggregate([
+        {
+          $match: {
+            email,
+          },
+        },
+        {
+          $unwind: '$enrolledIn',
+        },
+        {
+          $lookup: {
+            from: 'plans',
+            let: {
+              planId: '$enrolledIn.planId',
+              plans: '$enrolledIn',
+              progress: '$enrolledIn.progress',
+            },
+            pipeline: [
+              {
+                $match: { $expr: { $eq: ['$_id', '$$planId'] } },
+              },
+              {
+                $replaceRoot: {
+                  newRoot: {
+                    $mergeObjects: ['$$plans', '$$ROOT'],
+                  },
+                },
+              },
+              {
+                $project: {
+                  planId: 0,
+                },
+              },
+            ],
+            as: 'plans',
+          },
+        },
+        {
+          $group: {
+            _id: '$_id',
+            plans: { $push: { $first: '$plans' } },
+          },
+        },
+        {
+          $project: {
+            _id: 0,
+          },
+        },
+      ]);
+
+      const data = await cursor.toArray();
+
+      return data[0].plans;
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  static async findPlan({ email, planId }: { email: string; planId: string }) {
+    if (!this.userCollection) {
+      return;
+    }
+
+    try {
+      const cursor = await this.userCollection.aggregate([
+        {
+          $match: {
+            email,
+          },
+        },
+        {
+          $replaceRoot: {
+            newRoot: {
+              $first: {
+                $filter: {
+                  input: '$enrolledIn',
+                  as: 'item',
+                  cond: {
+                    $eq: ['$$item.planId', new mongodb.ObjectId(planId)],
+                  },
+                },
+              },
+            },
+          },
+        },
+      ]);
+
+      const plan = cursor.toArray();
+
+      return plan;
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  static async addPlan({ email, planId }: { email: string; planId: string }) {
+    if (!this.userCollection) {
+      return;
+    }
+
+    try {
+      const alreadyEnroled = await this.userCollection
```

Comment:
*alreadyEnrolled

not a big deal though, and line 279 naming is consistent

Reply:
yeah I misspelled the enrollment with 2 ts. My extension keeps complaining about that so I changed that to enrolment with 1 l.